dask/tests/test_highgraph.py

```diff
@@ -110,3 +111,33 @@ def plus_one(tasks):
 
     y.dask = dsk.map_tasks(plus_one)
     assert_eq(y, [42] * 3)
+
+
+def annot_map_fn(key):
+    return key[1:]
+
+
+@pytest.mark.parametrize(
+    "annotation",
+    [
+        {"worker": "alice"},
+        {"block_id": annot_map_fn},
+    ],
+)
+def test_single_annotation(annotation):
+    with dask.annotate(**annotation):
+        A = da.ones((10, 10), chunks=(5, 5))
+
+    alayer = A.__dask_graph__().layers[A.name]
+    assert alayer.annotations == annotation
+    assert dask.config.get("annotations", None) is None
```

Comment:
We might want to verify that a layer created outside of the context manager doesn't have annotations
```suggestion
    assert dask.config.get("annotations", None) is None

    b = A + 1
    assert not b.__dask_graph__().layers[b.name].annotations
```

Reply:
Currently, this produces a Blockwise object for the layer (not a BasicLayer)

```python
B = A + 1
assert isinstance(B.__dask_graph__().layers[B.name], Blockwise)
```

The  BasicLayer does live in Blockwise's _cached_dict attribute, but this feels like digging into the implementation a bit too far?

Reply:
I've added a ``test_no_annotations`` test case instead.

Reply:
I would hope that all `Layer` classes support annotations, not just `BasicLayer`

Reply:
Move annotation attachment to ``Layer.__init__``